admiral/pkg/clusters/handler.go

```diff
@@ -418,49 +418,51 @@ func handleVirtualServiceEvent(
 		}
 	}
 
-	dependentClusters := r.AdmiralCache.CnameDependentClusterCache.Get(virtualService.Hosts[0]).Copy()
-	if len(dependentClusters) > 0 {
-		for _, dependentCluster := range dependentClusters {
-			rc := r.GetRemoteController(dependentCluster)
-			if clusterId != dependentCluster {
-				log.Infof(LogFormat, "Event", "VirtualService", obj.Name, clusterId, "Processing")
-				if event == common.Delete {
-					err := rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(syncNamespace).Delete(ctx, obj.Name, v12.DeleteOptions{})
-					if err != nil {
-						if k8sErrors.IsNotFound(err) {
-							log.Infof(LogFormat, "Delete", "VirtualService", obj.Name, clusterId, "Either VirtualService was already deleted, or it never existed")
+	if virtualService.Hosts != nil {
```

Comment:
Its better to check the length of the slice, because if an empty `[]string` is assigned to `virtualService.Hosts` then it will not be nil.

Reply:
Ah, Right!!  I've updated